packages/asset-service/src/generateAssetData/avalanche/index.ts

```diff
@@ -1,10 +1,19 @@
 import { avalancheChainId } from '@shapeshiftoss/caip'
 
+import { getRenderedIdenticonBase64 } from '../../service/GenerateAssetIcon'
 import { avax } from '../baseAssets'
 import * as coingecko from '../coingecko'
 
 export const getAssets = async () => {
   const assets = await coingecko.getAssets(avalancheChainId)
   assets.push(avax)
-  return assets
+  return assets.map((asset) => {
+    if (!asset.icon) {
+      asset.icon = getRenderedIdenticonBase64(asset.assetId, asset.symbol, {
+        identiconImage: { size: 128, background: [45, 55, 72, 255] },
+        identiconText: { symbolScale: 7, enableShadow: true },
+      })
+    }
+    return asset
+  })
```

Comment:
I'd prefer a typed non-mutatey version of this:

```ts
export const getAssets = async (): Promise<Asset[]> => {
  const assets = await coingecko.getAssets(avalancheChainId)
  return [...assets, avax].map((asset) => ({
    ...asset,
    icon: asset.icon
      ? asset.icon
      : getRenderedIdenticonBase64(asset.assetId, asset.symbol, {
          identiconImage: { size: 128, background: [45, 55, 72, 255] },
          identiconText: { symbolScale: 7, enableShadow: true },
        }),
  }))
}
```

Reply:
https://github.com/shapeshift/lib/pull/1216/commits/405ed66ca234589240d65344fd4d05d0eeed6994